### PR TITLE
Ca 15 troop model

### PR DESCRIPTION
**Pull request by Kalloaa**

### CODE REVIEW

File: src/main/java/com/greenfoxacademy/chamaaragonite/troop/TroopFactory.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package com.greenfoxacademy.chamaaragonite.troop;
+
+import com.greenfoxacademy.chamaaragonite.exceptions.IncorrectTroopTypeException;
+import com.greenfoxacademy.chamaaragonite.troop.trooptypes.TroopType;
+
+public class TroopFactory {
+
+  //region methods
```
Comment (Elaiden) [inline src/main/java/com/greenfoxacademy/chamaaragonite/troop/TroopFactory.java]:
```suggestion
  // region Methods
```

File: src/main/java/com/greenfoxacademy/chamaaragonite/troop/TroopFactory.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  public static Troop createTroop(String type) throws IncorrectTroopTypeException {
+    try {
+      return TroopType.valueOf(type.toUpperCase()).create();
+    } catch (IllegalArgumentException e) {
+      throw new IncorrectTroopTypeException("Troop type doesn't exist");
+    }
+  }
+  //endregion methods
```
Comment (Elaiden) [inline src/main/java/com/greenfoxacademy/chamaaragonite/troop/TroopFactory.java]:
```suggestion
  // endregion Methods
```

File: src/main/java/com/greenfoxacademy/chamaaragonite/troop/TroopFactory.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+import com.greenfoxacademy.chamaaragonite.troop.trooptypes.TroopType;
+
+public class TroopFactory {
+
+  //region methods
+  public static Troop createTroop(String type) throws IncorrectTroopTypeException {
+    try {
+      return TroopType.valueOf(type.toUpperCase()).create();
+    } catch (IllegalArgumentException e) {
+      throw new IncorrectTroopTypeException("Troop type doesn't exist");
```
Comment (Elaiden) [inline src/main/java/com/greenfoxacademy/chamaaragonite/troop/TroopFactory.java]:
```suggestion
      throw new IncorrectTroopTypeException("Troop type doesn't exist!");
```

**Comment by ZaitzeV16** [comment]
Fixed typos in "fix typo" commit msg 😈

File: src/main/java/com/greenfoxacademy/chamaaragonite/troop/Troop.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  // region Fields
+  @Id
+  @GeneratedValue(strategy = GenerationType.IDENTITY)
+  private long id;
+
+  private int level = 1;
+  private int hp;
+  private int attack;
+  private int defense;
+
```
Comment (csadesz) [inline src/main/java/com/greenfoxacademy/chamaaragonite/troop/Troop.java]: Missing field, we should now when we created or updated the exact troop

**Pull request merged**